grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -1195,4 +1196,47 @@ class ProductService {
 
         return results.collect { new ProductSearchDto(it) }
     }
+
+    Product findProductByUpc(String upc){
```

Comment:
```
Product findProductByUpc(String upc) {
    return Product.findByUpc(upc)
}
```

We should actually have a more generic method like this but we can deal with that when we refactor the product lookup.
```
Product findProduct(String identifier){
    return Product.createCritieria.get {
        or {
            eq("id", identifier)
            eq("productCode", identifier)
            eq("upc", identifier)
            eq("ndc", identifier)
        } 
    }
}
```